content/docs/user-guide/project-structure/pipelines-files.md

```diff
@@ -0,0 +1,420 @@
+example:
+
+```yaml
+stages:
+  print2cols:
+    cmd: awk '{print $1 "\t" $2}' lines.txt > 2columns.txt
+    deps:
+      - lines.txt
+    outs:
+      - 2columns.txt
+```
+
+The most important part of a stage it's the terminal command(s) it executes
+(`cmd` field). This is what DVC runs when the stage is reproduced (see
+`dvc repro`).
+
+If a program executed by the command read input files, they can be defined as
+<abbr>dependencies</abbr> (`deps`). DVC will check whether they have changed to
+decide whether the stage requires re-execution (see `dvc status`).
+
+If the command writes files or directories, they can be defined as
+<abbr>outputs</abbr> (`outs`). DVC will track them going forward (similar to
+using `dvc add`).
+
+### Parameter dependencies
+
+[Parameters](/doc/command-reference/params) are a special type of stage
+dependency. They consist of a name/value pair to find in a YAML, JSON, TOML, or
+Python parameters file (`params.yaml` by default). Example:
+
+```yaml
+stages:
+  preprocess:
+    cmd: ./clean.sh raw.txt clean.txt
+    deps:
+      - raw.txt
+    params:
+      - threshold
+      - passes
+    outs:
+      - clean.txt
+```
+
+This allows several stages to depend on values of a shared structured file
+(which can be versioned directly with Git). See also `dvc params diff`.
+
+### Metrics and Plots outputs
+
+Like [common outputs](#outputs), <abbr>metrics</abbr> and <abbr>plots</abbr>
+files are produced by the stage `cmd`. However, their purpose is different.
+Typically they contain metadata to evaluate pipeline processes. Example:
+
+```yaml
+stages:
+  build:
+    cmd: python train.py
+    deps:
+      - features.csv
+    outs:
+      - model.pt
+    metrics:
+      - accuracy.txt:
+          cache: false
+    plots:
+      - auc.json:
+          cache: false
+```
+
+> `cache: false` is typical here, since they're small enough for Git to version
+> directly.
+
+The commands in `dvc metrics` and `dvc plots` help you display and compare
+metrics and plots.
+
+## Templating
+
+⚠️ This feature is only available in DVC 2.0 ⚠️
+
+`dvc.yaml` supports a templating format to insert values from different sources
+in the YAML structure itself. These sources can be
+[parameters files](/doc/command-reference/params), or `vars` defined in
+`dvc.yaml` instead.
+
+> Note that this parameterization feature is only supported via manual edition
+> of `dvc.yaml` and incompatible with `dvc run`.
+
+Let's say we have `params.yaml` (default params file) with the following
+contents:
+
+```yaml
+models:
+  us:
+    threshold: 10
+    filename: 'model-us.hdf5'
+```
+
+Those values can be used anywhere in `dvc.yaml` with the `${}` _substitution
+expression_:
+
+<!-- prettier-ignore-start -->
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python train.py
+      --thresh ${models.us.threshold}
+      --out ${models.us.filename}
+    outs:
+      - ${models.us.filename}:
+        cache: true
```

Comment:
```suggestion
          cache: true
```

Reply:
YAML blocks are auto-formatted in docs. A lot of them have strange indentation which I'm not sure would actually work 😆 Does it actually matter to DVC?

Reply:
Never mind, this change did pass tests 👍